opensrp-app/src/main/java/org/smartregister/commonregistry/CommonRepository.java

```diff
@@ -397,7 +436,7 @@ public void updateColumn(String tableName, ContentValues contentValues, String c
                 int columncount = cursor.getColumnCount();
                 HashMap<String, String> columns = new HashMap<String, String>();
                 for (int i = initialColumnCount; i < columncount; i++) {
-                    columns.put(additionalcolumns[i - initialColumnCount], cursor.getString(i));
+                    columns.put(additionalcolumns[i - initialColumnCount].getName(), cursor.getString(i));
```

Comment:
I think this might not be fetchable as String in cases where the column is not of type `VARCHAR`

Reply:
additionalcolums is from,  ColumnDetails[] additionalcolumns. ColumnDetails is a domain Object

Reply:
Most values are able to be returned as String, apart from Blob datatype. We can return the value as as `Object` but that would break implementation that assume the common person object stores Strings. Also I'm not sure if we have any plans to store Blobs in the database